tendrl/node_agent/manager/manager.py

```diff
@@ -100,19 +105,21 @@ def __init__(self, machine_id):
                     is None:
                 utils.delete_local_node_context()
 
+        node_id = utils.get_local_node_context()
         super(
             NodeAgentManager,
             self
         ).__init__(
             "node",
-            utils.get_local_node_context(),
-            utils.get_local_node_context(),
+            node_id,
             config,
             NodeAgentSyncStateThread(self),
             NodeAgentEtcdPersister(config),
-            "/tendrl_definitions_node-agent/data"
-        )
+            "/tendrl_definitions_node_agent/data",
+            node_id=node_id,
+            )
         self.register_node(machine_id)
+        self.load_and_execute_discovery_plugins(node_id)
```

Comment:
I think this is the right place because the manager is initialized and the node is registered to the etcd